app/src/main/java/com/example/findu/presentation/ui/extra/viewmodel/HomeExtraViewModel.kt

```diff
@@ -0,0 +1,139 @@
+package com.example.findu.presentation.ui.extra.viewmodel
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.findu.domain.model.extra.Center
+import com.example.findu.domain.model.extra.Department
+import com.example.findu.domain.model.extra.VolunteerWork
+import com.example.findu.presentation.model.HomeExtraContent
+import com.example.findu.presentation.type.HomeExtraButtonType
+import com.example.findu.presentation.type.view.LoadState
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.onStart
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+data class HomeExtraUiState(
+    val loadState: LoadState = LoadState.Idle,
+    val homeExtraButtonType: HomeExtraButtonType? = null,
+    val content: HomeExtraContent = HomeExtraContent.None
+
+)
+
+sealed class HomeExtraUiEvent {
+    data object LoadData : HomeExtraUiEvent()
+    data class SetHomeExtraType(val homeExtraButtonType: HomeExtraButtonType) : HomeExtraUiEvent()
+}
+
+sealed class HomeExtraUiEffect {
+
+}
+
+
+@HiltViewModel
+class HomeExtraViewModel @Inject constructor(
+    private val savedStateHandle: SavedStateHandle
+) : ViewModel() {
+
+    companion object {
+        private const val HOME_EXTRA_TYPE = "homeExtraType"
+    }
+
+
+    private val _uiState = MutableStateFlow(HomeExtraUiState())
+
+    private val _uiEffect = Channel<HomeExtraUiEffect>()
+    val uiEffect = _uiEffect.receiveAsFlow()
+
+    fun handleEvent(event: HomeExtraUiEvent) {
+        when (event) {
+            is HomeExtraUiEvent.LoadData -> loadData()
+            is HomeExtraUiEvent.SetHomeExtraType -> {
+                _uiState.value = _uiState.value.copy(homeExtraButtonType = event.homeExtraButtonType)
```

Comment:
```suggestion
                _uiState.update { it.copy(homeExtraButtonType = event.homeExtraButtonType) }
```

update 방식을 쓰는 건 어떨까요? 내부적으로 사용하는 `compareAndSet` 메소드가 Thread-safe 하게 동작되고, 같은 값인지 아닌지에 따라 판단해 만약 같은값을 사용한다면 불필요한 재연산을 하지 않는다고 합니다!
> This method is thread-safe and can be safely invoked from concurrent coroutines without external synchronization.